src/ign.cc

```diff
@@ -153,3 +159,113 @@ extern "C" SDFORMAT_VISIBLE int cmdPrint(const char *_path)
 
   return 0;
 }
+
+//////////////////////////////////////////////////
+extern "C" SDFORMAT_VISIBLE int cmdInertialStats(
+    const char *_path)
+{
+  if (!sdf::filesystem::exists(_path))
+  {
+    std::cerr << "Error: File [" << _path << "] does not exist.\n";
+    return -1;
+  }
+
+  sdf::Root root;
+  sdf::Errors errors = root.Load(_path);
+  if (!errors.empty())
+  {
+    for (auto &error : errors)
+    {
+      std::cerr << "Error: " << error.Message() << std::endl;
+    }
+  }
+
+  if (root.WorldCount() > 0)
+  {
+    std::cerr << "Error: Expected a model file but received a world file."
+            << std::endl;
+    return -1;
+  }
+
+  const sdf::Model *model = root.ModelByIndex(0);
+  if (!model)
+  {
+    std::cerr << "Error: Could not find the model." << std::endl;
+    return -1;
+  }
+
+  if (model->ModelCount() > 0)
+  {
+    std::cout << "Warning: Inertial properties of links in nested"
+            " models will not be included." << std::endl;
+  }
+
+  ignition::math::Inertiald totalInertial;
+
+  for (uint64_t i = 0; i < model->LinkCount(); i++)
+  {
+    ignition::math::Pose3d linkPoseRelativeToModel;
+    errors = model->LinkByIndex(i)->SemanticPose().
+      Resolve(linkPoseRelativeToModel, "__model__");
+
+    auto currentLinkInertial = model->LinkByIndex(i)->Inertial();
+    currentLinkInertial.SetPose(linkPoseRelativeToModel *
+      currentLinkInertial.Pose());
+    totalInertial += currentLinkInertial;
+  }
+
+  auto totalMass = totalInertial.MassMatrix().Mass();
+  auto xCentreOfMass = totalInertial.Pose().Pos().X();
+  auto yCentreOfMass = totalInertial.Pose().Pos().Y();
+  auto zCentreOfMass = totalInertial.Pose().Pos().Z();
+
+  std::cout << "Inertial statistics for model: " << model->Name() << std::endl;
+  std::cout << "---" << std::endl;
+  std::cout << "Total mass of the model: " << totalMass << std::endl;
+  std::cout << "---" << std::endl;
+
+  std::cout << "Centre of mass in model frame: " << std::endl;
+  std::cout << "X: " << xCentreOfMass << std::endl;
+  std::cout << "Y: " << yCentreOfMass << std::endl;
+  std::cout << "Z: " << zCentreOfMass << std::endl;
+  std::cout << "---" << std::endl;
+
+  std::cout << "Moment of inertia matrix: " << std::endl;
+
+  // Pretty print the MOI matrix
+  std::stringstream ss;
+  ss << totalInertial.Moi();
+
+  std::string s;
+  auto maxLength = 0u;
```

Comment:
there is a windows compiler warning on line 247. try `size_t` instead of `auto`?

* https://build.osrfoundation.org/job/sdformat-ci-pr_any-windows7-amd64/3612/msbuild/